expr/interval_compound_literal.go

```diff
@@ -0,0 +1,154 @@
+package expr
+
+import (
+	"errors"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalDateParts struct {
+	years      int32
+	months     int32
+	days       int32
+	seconds    int32
+	subSeconds int64
+}
```

Comment:
this structure represents value. proto carries type related information too (e.g. [Precision](https://github.com/substrait-io/substrait/blob/28025cbaa8dc3c65b736d8a68fa7070c465fb494/proto/substrait/algebra.proto#L924)). My intention is to construct value by value related fields.
We have another object to represent type so wanted to keep it separate so that separation of value/type remain clear

Reply:
Resolving this, since I feel (based on later comments) that we agree to not have proto to represent values